src/shared/mentionable_text_input.rs

```diff
@@ -198,21 +199,45 @@ live_design! {
     }
 }
 
+/// Subscriber adapter for MentionableTextInput
+/// This is used by parent components to subscribe to room member updates
+pub struct MentionableTextInputSubscriber {
+    widget_uid: WidgetUid,
+}
+
+/// Implement RoomMemberSubscriber trait to receive member update notifications
+impl RoomMemberSubscriber for MentionableTextInputSubscriber {
+    fn on_room_members_updated(
+        &mut self, cx: &mut Cx, room_id: &OwnedRoomId, members: Arc<Vec<RoomMember>>,
+    ) {
+        // Always send the update - we'll store the data regardless of current room
+        // This enables proper caching of room members for all rooms
+        log!(
+            "MentionableTextInputSubscriber({:?}) received members {} update for room {} in subscriber",
+            self.widget_uid,
+            members.len(),
+            room_id,
+        );
+        cx.action(MentionableTextInputAction::RoomMembersUpdated(room_id.clone(), members));
+    }
+}
+
 // /// A special string used to denote the start of a mention within
 // /// the actual text being edited.
 // /// This is used to help easily locate and distinguish actual mentions
 // /// from normal `@` characters.
 // const MENTION_START_STRING: &str = "\u{8288}@\u{8288}";
 
-
 /// Actions emitted by the MentionableTextInput component
```

Comment:
I think this comment is now (or always was) outdated/incorrect. These actions seem to be _handled_ by the MentionableTextInput widget, not emitted by it.

Reply:
Ok, upon a full search, it looks like some of them (RoomIdChanged) are indeed emitted by the MentionableTextInput widget, but most of the others are handled by it.

```suggestion
/// Actions emitted or hanlded by the MentionableTextInput widget.
```

In each variant of this enum, add docs that describe whether it's an action handled by or emitted by the MentionableTextInput widget.